wordle/user_interface.go

```diff
@@ -33,6 +34,10 @@ func NewWordleUI(ctx context.Context, wordleServ *Service, peerId string) *Wordl
 	}
 
 	wordleServ.SetLog(func(s string) {
+		// wait untill UI is initialilzed
+		for !ui.uiInitialized {
+			time.Sleep(200 * time.Millisecond)
+		}
```

Comment:
Instead of doing this, you can call SetLog when you know you can do it (UI is initialized) from the parent class.